### ISSUE TITLE
cue: export --out cue ignores -l, -p options

### ISSUE THREAD

**Issue by eonpatapon** [cue-lang/cue#1485]
**Is your feature request related to a problem? Please describe.**

I recently wanted to export a module data in a single cue file with cue export.

Turns out `-p` is ignored so no package declaration is added in the exported file.

Also `-l` is also ignored so it's not possible to export data at some path in the exported file.

I am aware that `-l` is used for importing data to cue in a specific path, but I think it should be possible to have this option for export as well.

Theses options are listed in `cue export --help`

**Describe the solution you'd like**

The `-p` option should have some effect when the output format is cue

The `-l` should have some effect for json, yaml... as well

**Additional context**

```
 cat x.cue 
package x

#X: string | int

foo: [string]: #X

foo: a: "a"
foo: b: 1

 cue export --out cue -p x -l '"x"'
foo: {
	a: "a"
	b: 1
}
```

I would expect:

```cue
package x

x: foo: {
  a: "a"
  b: 1
}
```

**Comment by jpluscplusm** [comment]
I believe `cue export` has respected the `--package` flag since https://cuelang.org/cl/1177546. The following test passes:

```
exec cue export .:foo --out cue --package bar
cmp stdout out
-- foo.cue --
package foo

x: 1
-- out --
package bar

x: 1
```

I've raised https://github.com/cue-lang/cue/issues/3671 to track that the `--package` flag currently appears to require a CUE package as input, rather than a file.

**Comment by myitcv** [comment]
@jpluscplusm is this issue otherwise complete?

**Comment by jpluscplusm** [comment]
@myitcv No; this tracks a request for both `-p` and `-l` to gain new `cue export` functionality, and my comment only highlights that the requested `-p` feature is available.

**Comment by mvdan** [comment]
The original reproducer as a testscript, and showing a change of package name:

```
exec cue export --out cue -p after -l "newpath"
cmp stdout stdout.golden

-- before.cue --
package before

#X: string | int

foo: [string]: #X

foo: a: "a"
foo: b: 1

-- stdout.golden --
package after

newpath: foo: {
	a: "a"
	b: 1
}
```
As @jpluscplusm mentions, the `-l` flag is the only one not obeyed at this point:

```
> exec cue export --out cue -p after -l "newpath"
[stdout]
package after

foo: {
	a: "a"
	b: 1
}
> cmp stdout stdout.golden
diff stdout stdout.golden
--- stdout
+++ stdout.golden
@@ -1,6 +1,6 @@
 package after
 
-foo: {
+newpath: foo: {
 	a: "a"
 	b: 1
 }
```

**Comment by mvdan** [comment]
Fixing `-p` with CUE inputs was https://review.gerrithub.io/c/cue-lang/cue/+/1177546 aka #2240.
